app/src/main/java/com/fstyle/structure_android/data/model/BaseModel.java

```diff
@@ -0,0 +1,16 @@
+package com.fstyle.structure_android.data.model;
+
+import com.google.gson.Gson;
+
+/**
+ * Created by le.quang.dao on 16/03/2017.
+ */
+
+public abstract class BaseModel<T> implements Cloneable {
+    public T data;
+
+    public BaseModel clone() {
```

Comment:
WARNING: [CloneThrowsCloneNotSupportedException] clone() method should throw CloneNotSupportedException
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/clone.html#CloneThrowsCloneNotSupportedException